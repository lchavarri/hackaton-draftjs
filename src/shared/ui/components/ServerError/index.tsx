import React from 'react';
import DismissableMessage from '../DismissableMessage';

// TODO: Add error message to Eidos Query for Kernel failure message.
export default function ServerError(props: any) {
  const { error } = props;
  if (!error) {
    return null;
  }

  const { message = 'Something went wrong', timestamp } = error;

  return (
    <DismissableMessage variant="negative" key={timestamp}>
      {message}
    </DismissableMessage>
  );
}
