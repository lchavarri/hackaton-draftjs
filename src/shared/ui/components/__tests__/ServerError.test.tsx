import React from 'react';
import { shallow, ReactWrapper } from 'enzyme';
import ServerError from '../ServerError';

describe('ServerError component', () => {
  let component: ReactWrapper;
  const props = {
    error: {
      message: 'Unable to get kernel'
    }
  };
  const getComponent = () => <ServerError {...props} />;
  const wrap = () => shallow(getComponent());

  beforeEach(() => {
    component = wrap();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should return null if no error prop was provided', () => {
    component.setProps({ error: null });
    component.update();
    expect(component.getElement()).toBe(null);
  });

  it('should render provided message if message is defined', () => {
    component.setProps({ error: { message: 'Some message' } });
    component.update();
    expect(component.html()).not.toContain('Something went wrong');
    expect(component.html()).toContain('Some message');
  });

  it('should render default message if message is not defined', () => {
    component.setProps({ error: {} });
    component.update();
    expect(component.html()).not.toContain(props.error.message);
    expect(component.html()).toContain('Something went wrong');
  });
});
