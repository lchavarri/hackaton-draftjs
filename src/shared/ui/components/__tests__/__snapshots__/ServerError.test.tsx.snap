// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerError component should match component snapshot 1`] = `
<DismissableMessage
  variant="negative"
>
  Unable to get kernel
</DismissableMessage>
`;
